src/session/content/message_row/reply.rs

```diff
@@ -0,0 +1,289 @@
+                }))
+                .bind(&*imp.photo_thumbnail, "visible", Some(message));
+            bindings.push(visibility_binding);
+        }
+    }
+}
+
+fn format_message_content_text(message: Message) -> String {
+    let session = message.chat().session();
+    match message.content().0 {
+        MessageContent::MessageText(content) => format_replied_message_text(content.text),
+        MessageContent::MessagePhoto(data) => {
+            if !data.caption.text.is_empty() {
+                // TODO: Replace "Photo" with photo thumbnail
+                format!(
+                    "[{}], {}",
+                    gettext("Photo"),
+                    format_replied_message_text(data.caption)
+                )
+            } else {
+                gettext("Photo")
+            }
+        }
+        MessageContent::MessageVideo(data) => {
+            // TODO: Add video thumbnail
+            if !data.caption.text.is_empty() {
+                format!(
+                    "[{}], {}",
+                    gettext("Video"),
+                    format_replied_message_text(data.caption)
+                )
+            } else {
+                gettext("Video")
+            }
+        }
+        MessageContent::MessageAudio(data) => {
+            format!("{} - {}", data.audio.performer, data.audio.title)
+        }
+        MessageContent::MessageAnimation(data) => {
+            if !data.caption.text.is_empty() {
+                format!(
+                    "[{}], {}",
+                    gettext("GIF"),
+                    format_replied_message_text(data.caption)
+                )
+            } else {
+                gettext("GIF")
+            }
+        }
+        MessageContent::MessageDocument(data) => data.document.file_name,
+        MessageContent::MessageSticker(data) => {
+            format!("{} {}", data.sticker.emoji, gettext("sticker"))
+        }
+        MessageContent::MessageDice(data) => data.emoji,
+        MessageContent::MessagePinMessage(data) => gettext!(
+            "Pinned {}",
+            item_row::stringify_pinned_message_content(
+                message.chat().history().message_by_id(data.message_id)
+            )
+        ),
+        MessageContent::MessageChatJoinByLink => gettext("joined the group via invite link"),
+        MessageContent::MessageChatJoinByRequest => gettext("joined to the group"),
+        MessageContent::MessageChatDeleteMember(data) => {
+            let user = session.user_list().get(data.user_id);
+            gettext!("removed {}", user.first_name())
+        }
+        MessageContent::MessageGame(data) => {
+            format!("🎮 {}", data.game.title)
+        }
+        MessageContent::MessagePoll(data) => data.poll.question,
+        _ => format!("<i>{}</i>", gettext("This message is unsupported")),
+    }
+}
+
+fn format_replied_message_text(text: FormattedText) -> String {
+    // TODO: change max text lenght depending on message content
```

Comment:
typo
I'm pretty sure Marcus or Marco will point out like 20 parts of my code that needs to be change so I'll wait and fix them all later :P